fly/fly_test.go

```diff
@@ -237,6 +237,41 @@ echo '[{"name":"abc"},{"name":"def"}]'
 			Expect(string(output)).To(Equal(expectedOutput))
 		})
 
+		Context("when optional vars are provided", func() {
+
+			var (
+				vars map[string]interface{}
+			)
+
+			BeforeEach(func() {
+				vars = map[string]interface{}{
+					"launch-missiles": true,
+					"credentials": map[string]string{
+						"username": "admin",
+						"password": "admin",
+					},
+				}
+			})
+
+			It("returns output without error", func() {
+				output, err := flyCommand.SetPipeline(pipelineName, configFilepath, nil, vars)
+				Expect(err).NotTo(HaveOccurred())
+
+				expectedOutput := fmt.Sprintf(
+					"%s %s %s %s %s %s %s %s %s %s %s %s\n",
+					"-t", target,
+					"set-pipeline",
+					"-n",
+					"-p", pipelineName,
+					"-c", configFilepath,
+					"-y", "launch-missiles=true",
+					"-y", "credentials=password: admin username: admin",
```

Comment:
is this correct? that syntax looks a little funky.

maybe use `json.Marshal` instead - it should be a bit more strict when it comes to complicated objects, and should work because all valid JSON is also valid YAML.

Reply:
Hi @vito I'm 85% sure this weirdness is the test code messing with the whitespace because it worked fine in tests I did using vars with multi-line YAML content, but I've updated it to JSON anyway just in case.